from PyQt5 import QtCore, QtGui, QtWidgets, uic
from PyQt5.QtWidgets import *
from PyQt5.QtWidgets import QMessageBox, QMainWindow, QStyledItemDelegate, QAction
import sys
from database import connect_database
from PyQt5.uic import loadUi
from source import admin


class UI_add_user(QMainWindow):
    def __init__(self, parent=None):
        super(UI_add_user, self).__init__()

    def setupUi(self):
        self.setObjectName("MainWindow")
        self.resize(603, 633)
        self.centralwidget = QtWidgets.QWidget(self)
        self.setWindowIcon(QtGui.QIcon("../icon/logo.png"))
        self.centralwidget.setObjectName("centralwidget")
        self.label_4 = QtWidgets.QLabel(self.centralwidget)
        self.label_4.setGeometry(QtCore.QRect(240, 0, 111, 41))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_4.setFont(font)
        self.label_4.setObjectName("label_4")
        self.label_5 = QtWidgets.QLabel(self.centralwidget)
        self.label_5.setGeometry(QtCore.QRect(70, 60, 101, 16))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.label_5.setFont(font)
        self.label_5.setObjectName("label_5")
        self.label_6 = QtWidgets.QLabel(self.centralwidget)
        self.label_6.setGeometry(QtCore.QRect(70, 110, 101, 16))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.label_6.setFont(font)
        self.label_6.setObjectName("label_6")
        self.label_7 = QtWidgets.QLabel(self.centralwidget)
        self.label_7.setGeometry(QtCore.QRect(70, 180, 101, 16))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.label_7.setFont(font)
        self.label_7.setObjectName("label_7")
        self.label_8 = QtWidgets.QLabel(self.centralwidget)
        self.label_8.setGeometry(QtCore.QRect(70, 240, 101, 16))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.label_8.setFont(font)
        self.label_8.setObjectName("label_8")
        self.label_9 = QtWidgets.QLabel(self.centralwidget)
        self.label_9.setGeometry(QtCore.QRect(70, 370, 101, 16))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.label_9.setFont(font)
        self.label_9.setObjectName("label_9")
        self.label_10 = QtWidgets.QLabel(self.centralwidget)
        self.label_10.setGeometry(QtCore.QRect(70, 430, 101, 16))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.label_10.setFont(font)
        self.label_10.setObjectName("label_10")
        self.txt_name = QtWidgets.QLineEdit(self.centralwidget)
        self.txt_name.setGeometry(QtCore.QRect(240, 50, 301, 31))
        self.txt_name.setObjectName("txt_name")
        self.txt_phone = QtWidgets.QLineEdit(self.centralwidget)
        self.txt_phone.setGeometry(QtCore.QRect(240, 170, 301, 31))
        self.txt_phone.setObjectName("txt_phone")
        self.txt_email = QtWidgets.QLineEdit(self.centralwidget)
        self.txt_email.setGeometry(QtCore.QRect(240, 230, 301, 31))
        self.txt_email.setObjectName("txt_email")
        self.tx_username = QtWidgets.QLineEdit(self.centralwidget)
        self.tx_username.setGeometry(QtCore.QRect(240, 360, 301, 31))
        self.tx_username.setObjectName("tx_username")
        self.txt_password = QtWidgets.QLineEdit(self.centralwidget)
        self.txt_password.setEchoMode(QLineEdit.Password)
        self.txt_password.setGeometry(QtCore.QRect(240, 420, 301, 31))
        self.txt_password.setObjectName("txt_password")
        self.date = QtWidgets.QDateEdit(self.centralwidget)
        self.date.setGeometry(QtCore.QRect(240, 110, 201, 31))
        self.date.setObjectName("date")
        self.btn_ok = QtWidgets.QPushButton(self.centralwidget)
        self.btn_ok.setGeometry(QtCore.QRect(140, 550, 89, 31))
        self.btn_ok.setStyleSheet("background-color:rgb(94, 186, 125)")
        self.btn_ok.setObjectName("btn_ok")
        self.btn_canncel = QtWidgets.QPushButton(self.centralwidget)
        self.btn_canncel.setStyleSheet("background-color:rgb(218, 91, 11)")
        self.btn_canncel.setGeometry(QtCore.QRect(350, 550, 89, 31))
        self.btn_canncel.setObjectName("btn_canncel")
        self.check_box_agree = QtWidgets.QCheckBox(self.centralwidget)
        self.check_box_agree.setGeometry(QtCore.QRect(240, 510, 251, 20))
        self.check_box_agree.setObjectName("check_box_agree")
        self.txt_email_2 = QtWidgets.QLineEdit(self.centralwidget)
        self.txt_email_2.setGeometry(QtCore.QRect(240, 290, 301, 31))
        self.txt_email_2.setText("")
        self.txt_email_2.setObjectName("txt_email_2")
        self.label_11 = QtWidgets.QLabel(self.centralwidget)
        self.label_11.setGeometry(QtCore.QRect(70, 300, 101, 16))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.label_11.setFont(font)
        self.label_11.setObjectName("label_11")
        self.label_12 = QtWidgets.QLabel(self.centralwidget)
        self.label_12.setGeometry(QtCore.QRect(70, 480, 101, 16))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.label_12.setFont(font)
        self.label_12.setObjectName("label_12")
        self.comboBox = QtWidgets.QComboBox(self.centralwidget)
        self.comboBox.setGeometry(QtCore.QRect(240, 471, 201, 31))
        self.comboBox.setObjectName("comboBox")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(self)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 603, 21))
        self.menubar.setObjectName("menubar")
        self.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(self)
        self.statusbar.setObjectName("statusbar")
        self.setStatusBar(self.statusbar)

        self.retranslateUi(self)
        # QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))
        self.label_4.setText(_translate("MainWindow", "ADD"))
        self.label_5.setText(_translate("MainWindow", "Name:"))
        self.label_6.setText(_translate("MainWindow", "Date of Birth:"))
        self.label_7.setText(_translate("MainWindow", "Phone:"))
        self.label_8.setText(_translate("MainWindow", "Email:"))
        self.label_9.setText(_translate("MainWindow", "Username:"))
        self.label_10.setText(_translate("MainWindow", "Password:"))
        self.btn_ok.setText(_translate("MainWindow", "Agree"))
        self.btn_canncel.setText(_translate("MainWindow", "Cancel"))
        self.check_box_agree.setText(_translate("MainWindow", "I agree to the Terms and Conditions"))
        self.label_11.setText(_translate("MainWindow", "Address:"))
        self.label_12.setText(_translate("MainWindow", "Role"))
        self.comboBox.setItemText(0, _translate("MainWindow", "admin"))
        self.comboBox.setItemText(1, _translate("MainWindow", "user"))

    # def ok(self):
    #     name = self.txt_name.text()
    #     dob = self.date.dateTime().toPyDateTime().date()
    #     phone = self.txt_phone.text()
    #     email = self.txt_email.text()
    #     address = self.txt_email_2.text()
    #     username = self.tx_username.text()
    #     password = self.txt_password.text()
    #     role = self.comboBox.currentText()
    #     if all(v is not '' for v in [name, dob, phone, email, address, username, password]):
    #         try:
    #             if self.check_box_agree.isChecked():
    #                 if connect_database.register(name, dob, phone, email, address, username, password):
    #                     QMessageBox.information(self, 'Login', 'dang ky thanh cong', QMessageBox.Close)
    #                     if (role == 'admin'):
    #                         connect_database.setAdmin(username)
    #                     self.hide()
    #             else:
    #                 QMessageBox.information(self, 'Login', 'nhap vao nut dong y di bro', QMessageBox.Close)
    #         except:
    #             QMessageBox.information(self, 'Login', 'dang ky that bai', QMessageBox.Close)
    #     else:
    #         QMessageBox.information(self, 'Login', 'nhap day du vao ', QMessageBox.Close)
    #
    # def cancel(self):
    #     self.hide()


if __name__ == "__main__":
    app = QtWidgets.QApplication(sys.argv)
    ui = UI_add_user()
    ui.setupUi()
    ui.show()
    sys.exit(app.exec_())
